 Kapowarr.py | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/Kapowarr.py b/Kapowarr.py
index 0e9da56..306976c 100644
--- a/Kapowarr.py
+++ b/Kapowarr.py
@@ -185,8 +185,13 @@ if __name__ == "__main__":
                 db_folder=db_folder
             )

-        except ValueError:
-            parser.error("The value for -d/--DatabaseFolder is not a folder")
+        except ValueError as e:
+            if e.args and e.args[0] == 'Database location is not a folder':
+                parser.error(
+                    "The value for -d/--DatabaseFolder is not a folder"
+                )
+            else:
+                raise e

     else:
         rc = Kapowarr()
--
2.48.1

